pkg/public/strategies.go

```diff
@@ -0,0 +1,209 @@
+package public
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/rds"
+	croAWS "github.com/integr8ly/cloud-resource-operator/pkg/providers/aws"
+	errorUtil "github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+	"time"
+)
+
+// reconciles aws strategy map, adding maintenance and backup window fields
+func reconcileAWSStrategyMap(ctx context.Context, client client.Client, tier, backupApplyOn, maintenanceApplyFrom, namespace string) error {
```

Comment:
Would there be an issue with expecting a time struct of some type here instead of a string for backupApplyOn and maintenanceApplyFrom?

Also, what do you think about changing the names to `backupTimeStart`, `maintenanceTimeStart` or something like that? To make it clear we're accepting the starting time and figuring out the rest